bank system.sql

```diff
@@ -1 +1,402 @@
+*/
+
+SELECT banks.name AS bank_name, full_name, accounts.balance AS account_balance, SUM(bank_cards.balance) AS cards_total, (accounts.balance - SUM(bank_cards.balance)) as balance_difference
+FROM accounts
+    JOIN bank_cards ON account_id = accounts.id
+    JOIN banks ON bank_id = banks.id
+    JOIN clients ON client_id = clients.id
+GROUP BY banks.name, full_name, accounts.balance
+HAVING accounts.balance - SUM(bank_cards.balance) <> 0;
+GO
+
+/* task 5
+Вывести кол-во банковских карточек для каждого соц статуса (2 реализации, GROUP BY и подзапросом)
+*/
+
+SELECT social_statuses.name AS social_status_name, COUNT(bank_cards.id) AS card_count
+FROM social_statuses
+    LEFT JOIN clients ON status_id = social_statuses.id
+    LEFT JOIN accounts ON client_id = clients.id
+    LEFT JOIN bank_cards ON account_id = accounts.id
+GROUP BY social_statuses.name
+ORDER BY 1;
+GO
+
+SELECT s.name AS social_status_name, 
+       (SELECT COUNT(*)
+        FROM bank_cards 
+            JOIN accounts ON account_id = accounts.id
+            JOIN clients ON client_id = clients.id
+            JOIN social_statuses ON status_id = social_statuses.id
+        WHERE social_statuses.id = s.id) AS card_count
+FROM social_statuses as s
+ORDER BY 1;
+GO
+
+/* task 6 
+Написать stored procedure которая будет добавлять по 10$ на каждый банковский аккаунт 
+для определенного соц статуса (У каждого клиента бывают разные соц. статусы. 
+Например, пенсионер, инвалид и прочее). 
+Входной параметр процедуры - Id социального статуса. 
+Обработать исключительные ситуации (например, был введен неверные номер соц. статуса. 
+Либо когда у этого статуса нет привязанных аккаунтов).
+*/
+
+CREATE PROC income_for_social_status @status_id INT
+AS
+DECLARE @max_id INT = 0;
+SELECT @max_id = MAX(id) FROM social_statuses;
+
+IF @status_id > @max_id 
+BEGIN
+    RAISERROR('Invalid index: Index greater than maximum', 16, 1);
+    RETURN;
+END
+IF @status_id < 1 
+BEGIN
+    RAISERROR('Invalid index: Index less than 1', 16, 1);
+    RETURN;
+END
+
+IF NOT EXISTS(
+    SELECT *
+    FROM accounts
+    WHERE client_id IN (SELECT id 
+                        FROM clients 
+                        WHERE status_id = @status_id))
+BEGIN
+    RAISERROR('Invalid index: Theres no accounts for that social status id', 16, 1);
+    RETURN;
+END
+
+UPDATE accounts 
+    SET balance += 10
+WHERE client_id IN (SELECT id 
+                    FROM clients 
+                    WHERE status_id = @status_id);
+GO
+
+SELECT client_id, status_id, social_statuses.name AS social_status_name, balance AS account_balance
+FROM accounts
+    JOIN clients ON client_id = clients.id
+    JOIN social_statuses ON status_id = social_statuses.id;
+GO
+
+EXEC income_for_social_status 5;
+GO
+
+SELECT client_id, status_id, social_statuses.name AS social_status_name, balance AS account_balance
+FROM accounts
+    JOIN clients ON client_id = clients.id
+    JOIN social_statuses ON status_id = social_statuses.id;
+GO
```

Comment:
The task is not working, the balance has not changed
![image](https://user-images.githubusercontent.com/33979910/202870677-acf21a75-a965-4e95-a26f-f6b725be60a9.png)

Reply:
I don't really understand what the problem is(
I've already checked it on two computers and everything works for me. I don't see any errors in the body of the procedure either.
![image](https://user-images.githubusercontent.com/72939751/202872587-b6ccd9b4-6ece-425a-a652-ca813f4f94d2.png)

Maybe you accidentally deleted something? Please try again to delete the database and run the script.